code/player/TTTPlayer.Spectating.cs

```diff
@@ -0,0 +1,52 @@
+using System.Collections.Generic;
+
+using Sandbox;
+
+using TTTReborn.Globals;
+using TTTReborn.Player.Camera;
+
+namespace TTTReborn.Player
+{
+    public partial class TTTPlayer
+    {
+        private int _targetIdx = 0;
+
+        [Event("tttreborn.player.died")]
+        private static void OnPlayerDied(TTTPlayer deadPlayer)
+        {
+            if (!Host.IsClient || Local.Pawn is not TTTPlayer player)
+            {
+                return;
+            }
+
+            if (player.IsSpectatingPlayer && player.CurrentPlayer == deadPlayer)
+            {
+                player.UpdateObservatedPlayer();
+            }
+        }
+
+        public void UpdateObservatedPlayer()
```

Comment:
Do we want to instead go with `Spectated` instead of `Observated`?

Reply:
I guess technically it could be specific to `Observation`, up to you, both are fine with me.

Reply:
Maybe in another PR to change every `observate` into `spectate`. I think I've choosen `observate` as there have been some issues using `spectate` I can't remember currently.

Reply:
np, I'm fine with both.